src/action.ts

```diff
@@ -53,8 +53,25 @@ export async function run(
 				core.info('Release is already finalized so skipping.');
 				return;
 			}
-			// Finalize release and done!
-			return await balena.finalize(previousRelease.id);
+
+			// Finalize release!
+			await balena.finalize(previousRelease.id);
+
+			rawVersion = await balena.getReleaseVersion(previousRelease.id);
+
+			if (inputs.createTag && rawVersion) {
+				try {
+					await github.createTag(repoContext, rawVersion);
+				} catch (e: any) {
+					if (e.message !== 'Reference already exists') {
+						throw e;
+					}
+					core.info('Git reference already exists.');
+					return;
+				}
+			}
```

Comment:
I'm not a huge fan of the duplicate code but we can merge it since I plan to refactor the action file soon to reduce some of the branches it has going on.